src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+        log.info("Starting suggest api step");
+
+        // Create suggest request for interval, history, window_delay
+        SuggestConfigParamRequest suggestRequest = new SuggestConfigParamRequest(
+            AnalysisType.AD,
+            detector,
+            "interval,history,window_delay",
+            TimeValue.timeValueSeconds(SUGGEST_API_TIMEOUT_SECONDS)
+        );
+
+        adClient.suggestAnomalyDetector(suggestRequest, new ActionListener<SuggestConfigParamResponse>() {
+            @Override
+            public void onResponse(SuggestConfigParamResponse response) {
+                try {
+                    AnomalyDetector optimizedDetector = applySuggestionsToDetector(detector, response);
+                    validateModelPhase(optimizedDetector, tenantId, maxRetries, 0, listener);
+
+                } catch (Exception e) {
+                    validateModelPhase(detector, tenantId, maxRetries, 0, listener);
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // Continue to model validation with original detector even if suggest fails
+                validateModelPhase(detector, tenantId, maxRetries, 0, listener);
+            }
+        });
+    }
+
+    // Checks if there's enough data to train the model (final validation before creating detector)
+    private <T> void validateModelPhase(
+        AnomalyDetector detector,
+        String tenantId,
+        int maxRetries,
+        int currentRetry,
+        ActionListener<T> listener
+    ) {
+        log.info("Starting model validation");
+
+        callValidationAPI(detector, "model", new ActionListener<ValidateConfigResponse>() {
+            @Override
+            public void onResponse(ValidateConfigResponse response) {
+
+                if (response.getIssue() != null) {
+                    String issueAspect = response.getIssue().getAspect().toString();
+                    boolean isBlockingError = issueAspect != null && issueAspect.toLowerCase(Locale.ROOT).startsWith("detector");
+                    String errorMessage = response.getIssue().getMessage();
+                    if (currentRetry < MAX_MODEL_VALIDATION_RETRIES) {
+                        retryModelValidation(detector, errorMessage, tenantId, maxRetries, currentRetry + 1, listener);
+                    } else {
+                        // Max retries reached
+                        if (isBlockingError) {
+                            log.error("Max retries reached with blocking validation error: {}", errorMessage);
+                            respondWithError(listener, detector.getIndices().get(0), "validation", errorMessage);
+                        } else {
+                            DetectorResult result = DetectorResult
+                                .failedValidation(detector.getIndices().get(0), "Non-blocking warning: " + errorMessage);
+                            listener.onResponse((T) result.toJson());
```

Comment:
If `detector.getIndices().get(0)` throws an exception the listener will never be completed. Probably need some exception handling here.